File: Program's_Contributed_By_Contributors/C++/Quick_Sort.cpp
// C++ program for implementation of Quick sort
#include <bits/stdc++.h>
using namespace std;

// Partition function
int partition(int arr[], int low, int high)
{
    int pivot = arr[high]; // pivot
    int i = (low - 1); // Index of smaller element

    for (int j = low; j <= high - 1; j++) {
        // If current element is smaller than or
        // equal to pivot
        if (arr[j] <= pivot) {
            i++; // increment index of smaller element
            swap(arr[i], arr[j]);
        }
    }
    swap(arr[i + 1], arr[high]);
    return (i + 1);
}

// A function to implement quick sort
// Time Complexity: O(nlogn)
void quickSort(int arr[], int low, int high)
{
	if (low < high)
	{       
		// pi is partitioning index, arr[p] is now
        // at right place
        int pi = partition(arr, low, high);

        // Separately sort elements before
        // partition and after partition
        quickSort(arr, low, pi - 1);
        quickSort(arr, pi + 1, high);
	}
}

// Function to print an array
void printArray(int arr[], int size)
{
	int i;
	for (i = 0; i < size; i++)
		cout << arr[i] << " ";
	cout << endl;
}

// Driver code
int main()
{
	int arr[] = { 5, 1, 4, 2, 8};
	int N = sizeof(arr) / sizeof(arr[0]);
	quickSort(arr, N);
	cout << "Sorted array: \n";
	printArray(arr, N);
	return 0;
}

File: Program's_Contributed_By_Contributors/QuickSort.cs
using System;

class QuickSort
{
    public static void Sort(int[] arr)
    {
        QuickSortRecursive(arr, 0, arr.Length - 1);
    }

    private static void QuickSortRecursive(int[] arr, int left, int right)
    {
        if (left < right)
        {
            int pivotIndex = Partition(arr, left, right);
            QuickSortRecursive(arr, left, pivotIndex - 1);
            QuickSortRecursive(arr, pivotIndex + 1, right);
        }
    }

    private static int Partition(int[] arr, int left, int right)
    {
        int pivot = arr[right];
        int i = left - 1;

        for (int j = left; j < right; j++)
        {
            if (arr[j] <= pivot)
            {
                i++;
                Swap(arr, i, j);
            }
        }

        Swap(arr, i + 1, right);
        return i + 1;
    }

    private static void Swap(int[] arr, int i, int j)
    {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    public static void Main(string[] args)
    {
        int[] arr = { 12, 4, 5, 6, 7, 3, 1, 15 };

        Console.Write("Original Array: ");
        PrintArray(arr);

        Sort(arr);

        Console.Write("Sorted Array: ");
        PrintArray(arr);
    }

    private static void PrintArray(int[] arr)
    {
        foreach (var item in arr)
        {
            Console.Write(item + " ");
        }
        Console.WriteLine();
    }
}
